### PR TITLE
Polls rework2

### LINKED ISSUE #413: Backend /api/polls/:id/submit Endpoint Work
**Please describe what has to be done**
Add in the ability to have logged in users submit poll answers. Resubmissions should overwrite the previous submission answers (not the entire submission). This also needs more testing and error checking throughout it. 

**Additional context**
See #393

### PR DESCRIPTION
**Pull request by Isaac-the-Man**
### Description of change(s), including why:

Reworked route `/api/polls/:id/submit`. Added a helper function `isLogedIn()` and a middleware `checkPollPublic()` to `utils.js`. Updated eslint to support emca 2018.

Documentation added:
- [new route](https://github.com/PollBuddy/PollBuddy/wiki/Specifications-%E2%80%90-Backend-Routes-(Polls)#post-idsubmit)
- [isLoggedIn](https://github.com/PollBuddy/PollBuddy/wiki/Specifications-%E2%80%90-Backend-Overview#isloggedinreq)
- [checkPollPublic](https://github.com/PollBuddy/PollBuddy/wiki/Specifications-%E2%80%90-Backend-Overview#checkpollpublic)

### Type of change(s)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Any work to still do or other concerns:

- Needs testing for logged in user once this is implemented in the frontend.
- Polls should implement the new `Public` attribute in the backend.

### Closing issues:

- Closes #413

### CODE REVIEW

File: PollBuddy-Server/backend/modules/utils.js [hunk 1]
```diff
@@ -40,6 +40,37 @@ async function validateID(collection, id) {
...
+async function checkPollPublic(req, res, next) {
+  // validate poll id
+  const id = await validateID("polls", req.params.id);
+  if (!id) {
+    return res.status(400).send(createResponse(null, "Invalid ID."));
+  }
+  req.parsedID = id;
+  // check poll publicity
+  try {
+    const poll = await mongoConnection.getDB().collection("polls").findOne({"_id": id});
```
Comment (hbh7) [inline PollBuddy-Server/backend/modules/utils.js:60]: This should probably have a filter to only return that one property for efficiency.

Comment (hbh7) [reply PollBuddy-Server/backend/modules/utils.js:60]: I think you just add `, { Public: true }` after the `{"_id": id}` but I'm not 100% certain.

File: PollBuddy-Server/backend/routes/polls.js [hunk 2]
```diff
@@ -90,84 +90,63 @@ router.post("/:id/edit", async (req, res) => {
...
+ * Submit/re-submit poll answer.
+ * Depending on the "Pubic" attribute of the poll it might requires sign-in.
+ * @typedef {Object} Answers
+ * @property {string} QuestionID - ID of the question.
+ * @property {string} Answer - Answer response, could be null or empty string.
+ * @postdata {Answers[]} payload
+ * @throws 400 - Invalid ID.
+ * @throws 403 - Sign-In required.
+ * @throws 500 - An error occurred while reading the database.
+ * @throws 500 - An error occurred while writing to the database.
```
Comment (hbh7) [inline PollBuddy-Server/backend/routes/polls.js:103]: I know we don't have a clear defined format for this, but could you combine the two database messages into `An error occurred while communicating with the database.`? It's not really relevant to report to the frontend which database action failed, just that a database issue occurred and therefore the task failed. That would simplify using the endpoint too a bit. 

I would like to make that a standard format though. Thoughts from anyone?

**Review by hbh7 — CHANGES_REQUESTED** [review]
Just a few small changes, everything else looks good!

**Review by hbh7 — APPROVED** [review]
Great! :)

**Review by neha-deshpande001 — APPROVED** [review]

**Pull request merged**